lib/cli/src/commands/package/unpack.rs

```diff
@@ -0,0 +1,90 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+
+/// Extract contents of a container to a directory.
+#[derive(clap::Parser, Debug)]
+pub struct CmdPackageUnpack {
+    /// The output directory.
+    #[clap(short = 'o', long)]
+    out_dir: PathBuf,
+
+    /// Overwrite existing directories/files.
+    #[clap(long)]
+    overwrite: bool,
+
+    /// Path to the package.
+    package_path: PathBuf,
+}
+
+impl CmdPackageUnpack {
+    pub(crate) fn execute(&self) -> Result<(), anyhow::Error> {
+        eprintln!("Unpacking...");
+
+        let pkg = webc::compat::Container::from_disk(&self.package_path).with_context(|| {
+            format!(
+                "could not open package at '{}'",
+                self.package_path.display()
+            )
+        })?;
+
+        let outdir = &self.out_dir;
+        std::fs::create_dir_all(outdir)
+            .with_context(|| format!("could not create output directory '{}'", outdir.display()))?;
+
+        pkg.unpack(outdir, self.overwrite)
+            .with_context(|| "could not extract package".to_string())?;
+
+        eprintln!("Extracted package contents to '{}'", self.out_dir.display());
```

Comment:
You can use `indicatif::ProgressBar`'s `finish_with_message()` for this bit.

Reply:
Let's do this in a followup.